### PR TITLE
Dev tools plugin

### PR DESCRIPTION
**Pull request by jreyno77**

### CODE REVIEW

**Review by rob-reynolds — COMMENTED** [review]
Add plugin to the idea config so that you get those features:
compiler.xml
encodings.xml

**Comment by rob-reynolds** [comment]
I'd recommend updating the name to dev-tools.  "dev" suggests it's for the development environment.

**Comment by rob-reynolds** [comment]
Use the new Description annotation to expose the documentation.

File: plugin/dev/src/main/java/org/opencds/cqf/ruler/plugin/dev/utils/DevUtilities.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package org.opencds.cqf.ruler.plugin.dev.utils;
```
Comment (rob-reynolds) [inline plugin/dev/src/main/java/org/opencds/cqf/ruler/plugin/dev/utils/DevUtilities.java:1]: I don't think this is necessary

File: plugin/dev/src/main/java/org/opencds/cqf/ruler/plugin/dev/utils/IDevUtilities.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package org.opencds.cqf.ruler.plugin.dev.utils;
+
```
Comment (rob-reynolds) [inline plugin/dev/src/main/java/org/opencds/cqf/ruler/plugin/dev/utils/IDevUtilities.java:2]: Don't think this is necessary either.  Just extend both in your class

File: plugin/dev/src/main/resources/application.yaml [hunk 3]
```diff
@@ -0,0 +1,191 @@
+debug: true
+
```
Comment (rob-reynolds) [inline plugin/dev/src/main/resources/application.yaml:2]: I don't think you need all this configuration in the plugin.  The plugin should only specify the stuff specific to it.

File: plugin/dev/src/test/java/org/opencds/cqf/ruler/plugin/dev/IServerSupport.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
+package org.opencds.cqf.ruler.plugin.dev;
```
Comment (rob-reynolds) [inline plugin/dev/src/test/java/org/opencds/cqf/ruler/plugin/dev/IServerSupport.java:1]: Use / move this to the utility plugin

File: plugin/dev/src/test/java/org/opencds/cqf/ruler/plugin/dev/TesterUtilities.java [hunk 5]
```diff
@@ -0,0 +1,8 @@
+package org.opencds.cqf.ruler.plugin.dev;
```
Comment (rob-reynolds) [inline plugin/dev/src/test/java/org/opencds/cqf/ruler/plugin/dev/TesterUtilities.java:1]: Use / move this to the test-utility plugin.

File: plugin/dev/src/test/java/org/opencds/cqf/ruler/plugin/dev/dstu3/CacheValueSetsProviderIT.java [hunk 6]
```diff
@@ -0,0 +1,224 @@
...
+    
+    @Autowired
+    private DaoRegistry myDaoRegistry;
+    
+    @LocalServerPort
+    private int port;
+
+    @Test
+    public void testCacheValueSetsEndpointDNE() throws Exception {
+        Endpoint endpoint = new Endpoint();
```
Comment (rob-reynolds) [inline plugin/dev/src/test/java/org/opencds/cqf/ruler/plugin/dev/dstu3/CacheValueSetsProviderIT.java:62]: Format tests like:

//Setup

//Execute

//Assertion

File: plugin/dev/src/test/resources/org/opencds/cqf/ruler/plugin/dev/dstu3/endpoint/LocalServerEndpoint.json [hunk 7]
```diff
@@ -0,0 +1,18 @@
+{
```
Comment (rob-reynolds) [inline plugin/dev/src/test/resources/org/opencds/cqf/ruler/plugin/dev/dstu3/endpoint/LocalServerEndpoint.json:1]: Move test resources to the test/resources directory.

**Review by rob-reynolds — CHANGES_REQUESTED** [review]
See all the comments

**Comment by rob-reynolds** [comment]
Compiles locally.
Some tests did not pass locally.

File: plugin/dev/src/test/java/org/opencds/cqf/ruler/plugin/dev/TesterUtilities.java (hunk 5 above)
Comment (jreyno77) [reply plugin/dev/src/test/java/org/opencds/cqf/ruler/plugin/dev/TesterUtilities.java]: I removed it like the other unnecessary Utilities

**Review by JPercival — APPROVED** [review]

**Pull request merged**